modules/rgbd/src/tsdf.cpp

```diff
@@ -2,81 +2,35 @@
 // of this distribution and at http://opencv.org/license.html
 
-// This code is also subject to the license terms in the LICENSE_KinectFusion.md file found in this module's directory
+// This code is also subject to the license terms in the LICENSE_KinectFusion.md file found in this
+// module's directory
 
-#include "precomp.hpp"
 #include "tsdf.hpp"
-#include "opencl_kernels_rgbd.hpp"
-
-namespace cv {
 
-namespace kinfu {
+#include "opencl_kernels_rgbd.hpp"
+#include "precomp.hpp"
 
-// TODO: Optimization possible:
-// * volumeType can be FP16
-// * weight can be int16
-typedef float volumeType;
-struct Voxel
+namespace cv
 {
-    volumeType v;
-    int weight;
-};
-typedef Vec<uchar, sizeof(Voxel)> VecT;
-
-
-class TSDFVolumeCPU : public TSDFVolume
+namespace kinfu
 {
-public:
-    // dimension in voxels, size in meters
-    TSDFVolumeCPU(Point3i _res, float _voxelSize, cv::Affine3f _pose, float _truncDist, int _maxWeight,
-                  float _raycastStepFactor, bool zFirstMemOrder = true);
-
-    virtual void integrate(InputArray _depth, float depthFactor, cv::Affine3f cameraPose, cv::kinfu::Intr intrinsics) override;
-    virtual void raycast(cv::Affine3f cameraPose, cv::kinfu::Intr intrinsics, cv::Size frameSize,
-                         cv::OutputArray points, cv::OutputArray normals) const override;
-
-    virtual void fetchNormals(cv::InputArray points, cv::OutputArray _normals) const override;
-    virtual void fetchPointsNormals(cv::OutputArray points, cv::OutputArray normals) const override;
-
-    virtual void reset() override;
-
-    volumeType interpolateVoxel(cv::Point3f p) const;
-    Point3f getNormalVoxel(cv::Point3f p) const;
-
-#if USE_INTRINSICS
-    volumeType interpolateVoxel(const v_float32x4& p) const;
-    v_float32x4 getNormalVoxel(const v_float32x4& p) const;
-#endif
-
-    // See zFirstMemOrder arg of parent class constructor
-    // for the array layout info
-    // Consist of Voxel elements
-    Mat volume;
-};
-
-
-TSDFVolume::TSDFVolume(Point3i _res, float _voxelSize, Affine3f _pose, float _truncDist, int _maxWeight,
-                       float _raycastStepFactor, bool zFirstMemOrder) :
-    voxelSize(_voxelSize),
-    voxelSizeInv(1.f/_voxelSize),
-    volResolution(_res),
-    maxWeight(_maxWeight),
-    pose(_pose),
-    raycastStepFactor(_raycastStepFactor)
+TSDFVolume::TSDFVolume(float _voxelSize, Affine3f _pose, float _raycastStepFactor, float _truncDist,
+                       int _maxWeight, Point3i _resolution, bool zFirstMemOrder)
+    : Volume(_voxelSize, _pose, _raycastStepFactor),
+      volResolution(_resolution),
+      maxWeight(_maxWeight)
 {
     // Unlike original code, this should work with any volume size
     // Not only when (x,y,z % 32) == 0
-
-    volSize = Point3f(volResolution) * voxelSize;
-
-    truncDist = std::max(_truncDist, 2.1f * voxelSize);
+    volSize   = Point3f(volResolution) * voxelSize;
+    truncDist = std::max(_truncDist, 4.0f * voxelSize);
```

Comment:
Considering that this class is still not exposed: Actually this affects HashTSDFVolume since the HashTSDF uses TSDFVolume units. So, it would be better to keep this as 4.0f, although in **my opinion**, this is not required at all.